opm/grid/common/ZoltanPartition.cpp

```diff
@@ -176,16 +137,239 @@ zoltanGraphPartitionGridOnRoot(const CpGrid& cpgrid,
+        float ver = 0;
+
+        rc = Zoltan_Initialize(argc, argv, &ver);
+        zz = Zoltan_Create(MPI_COMM_SELF);
+        if ( rc != ZOLTAN_OK )
+        {
+            OPM_THROW(std::runtime_error, "Could not initialize Zoltan!");
+        }
+
+        setDefaultZoltanParameters(zz);
+        Zoltan_Set_Param(zz, "NUM_GLOBAL_PARTS", std::to_string(cc.size()).c_str());
+
+        // For the load balancer one process has the whole grid and
+        // all others an empty partition before loadbalancing.
+        bool partitionIsEmpty     = cc.rank()!=root;
+
+        if( wells )
+        {
+            Zoltan_Set_Param(zz,"EDGE_WEIGHT_DIM","1");
+            gridAndWells.reset(new CombinedGridWellGraph(cpgrid,
+                                                           wells,
+                                                           transmissibilities,
+                                                           partitionIsEmpty,
+                                                           edgeWeightsMethod));
+            Dune::cpgrid::setCpGridZoltanGraphFunctions(zz, *gridAndWells,
+                                                        partitionIsEmpty);
+        }
+        else
+        {
+            Dune::cpgrid::setCpGridZoltanGraphFunctions(zz, cpgrid, partitionIsEmpty);
+        }
+
+        rc = Zoltan_LB_Partition(zz, /* input (all remaining fields are output) */
+                                 &changes,        /* 1 if partitioning was changed, 0 otherwise */
+                                 &numGidEntries,  /* Number of integers used for a global ID */
+                                 &numLidEntries,  /* Number of integers used for a local ID */
+                                 &numImport,      /* Number of vertices to be sent to me */
+                                 &importGlobalGids,  /* Global IDs of vertices to be sent to me */
+                                 &importLocalGids,   /* Local IDs of vertices to be sent to me */
+                                 &importProcs,    /* Process rank for source of each incoming vertex */
+                                 &importToPart,   /* New partition for each incoming vertex */
+                                 &numExport,      /* Number of vertices I must send to other processes*/
+                                 &exportGlobalGids,  /* Global IDs of the vertices I must send */
+                                 &exportLocalGids,   /* Local IDs of the vertices I must send */
+                                 &exportProcs,    /* Process to which I send each of the vertices */
+                                 &exportToPart);  /* Partition to which each vertex will belong */
+
+        // This is very important: by default, zoltan sets numImport to -1,
+        // as we are only running Zoltan in serial.
+        // In order to make sense of the rest of  the code, this must be set
+        // to 0.
+        numImport = 0;
+
+
+    }
+    MPI_Barrier(cc);
```

Comment:
In this version this is not need as the latter scatter is a global synchronization.

Reply:
This can be removed, it was only put in for debugging and then I forgot to remove it.